dist/joii.js

```diff
@@ -1329,8 +1685,43 @@ JOII.InterfaceBuilder = function() {
                 verifyMeta(this, p1, p2, 'Method');
 
                 // Verify function signature.
-                if (p1.getParameters().length !== p2.getParameters().length) {
-                    throw 'Method ' + p1.getName() + ' does not match the parameter count as defined in the interface ' + this.name + '.';
+                var argsInterface = p1.getParameters();
+                var argsClass = p2.getParameters();
+
+                if (argsInterface.length == 0 || typeof (argsInterface[0]) !== 'object') {
+                    // fallback for backwards compatibility
+                    if (p1.getParameters().length !== p2.getParameters().length) {
+                        throw 'Method ' + p1.getName() + ' does not match the parameter count as defined in the interface ' + this.name + '.';
+                    }
+                }
```

Comment:
Could you please explain the issue with this particular area for code style? I can't see anything wrong with it.